pgrollup/parser_pg.py

```diff
@@ -0,0 +1,290 @@
+    '''
+    def string(self, s):
+        '''
+        Example input:
+        [Tree('string', [Token('ESCAPED_STRING', '"t1"')]),Tree('string', [Token('ESCAPED_STRING', '"num"')])]
+        Example output:
+        ['t1','num']
+       '''
+        (s,) = s
+        return s[1:-1]
+    def number(self, n):
+        '''
+        Example input:
+        Tree('number', [Token('SIGNED_NUMBER', '40')])
+        Example output:
+        40
+        '''
+        (n,) = n
+        return int(n)                        
+    def pair(self,key_value):
+        k, v = key_value
+        return str(k),v
+                       
+    list = list   
+#   pair = tuple
+    dict = dict
+
+    #NAME = lambda self, n: n
+    name  = lambda self,n: ('TYPE',str(n[0]))
+    null = lambda self, _: None
+    true = lambda self, _: True
+    false = lambda self, _: False
+
+parser = Lark(grammar, start='value', lexer='standard')
+
+import pprint
+def parse(text):
+    '''
+    >>> pprint.pprint(parse(sql0))
+    '''
+    tree = parser.parse(text)
+    return Transformer().transform(tree)
+
+def parse_tree(text):
+    print(parser.parse(text).pretty())
+
+################################################################################
+# internal helper functions
+################################################################################
+
+def expr_to_str(expr):
+    '''
+    This function tramsforms expressions into strings.
+    
+    Example input:
+    {'TYPE': 'AEXPR',
+    'lexpr': {'TYPE': 'COLUMNREF', 'fields': ['test'], 'location': 80},
+    'location': 84,
+    'name': ['>='],
+    'rexpr': {'TYPE': 'COLUMNREF', 'fields': ['from'], 'location': 86}}
+    Exmpale output:
+    'test>=from'
+    '''
+    if expr['TYPE']=='A_CONST':
+        return str(expr['val'])
+    elif expr['TYPE']=='COLUMNREF':
+        return str(expr['fields'][0])
+    elif expr['TYPE']=='FUNCCALL':
+        if expr['agg_star']==True:
+            return expr['funcname'][0]+'(*)'
+        else:
+            return expr['funcname'][0]+'('+expr_to_str(expr['args'][0])+')'
+#           else:
+#               return expr['funcname'][0]+'('+expr['args'][0]['fields'][0]+')'
+    elif expr['TYPE']=='AEXPR':
+        return '('+expr_to_str(expr['lexpr']) + expr['name'][0] + expr_to_str(expr['rexpr'])+')'
+
+def _gettype(info):
+    '''
+    This function returns a query's join type
+    '''
+    if info['jointype'] == 0:
+        return "INNER JOIN"
+    elif info['jointype'] ==1:
+        return "OUTER JOIN"
+    elif info['jointype'] ==2:
+        return "FULL JOIN"
+
+def _getalias(info):
+    '''
+    This function returns any aliases for a selected column
+    '''
+    if info['name'] != None:
+        return info['name'][1]
+    else:
+        return ''
+
+def _getgroups(sql):
+
+    if sql[0]['stmt']['groupClause'] != None:
+        return [[x['fields'][0],str(x['fields'][0])] for x in sql[0]['stmt']['groupClause']]
+    else:
+        return ''
+def _getjoins(info):
```

Comment:
According to PEP8, there should be 2 empty lines between all functions; none of your current functions follow this guideline, but this case is particularly egregious since there are no spaces